src/Core/Silk.NET.BuildTools/Converters/Readers/OpenCLReader.cs

```diff
@@ -835,42 +755,181 @@ public IEnumerable<Enum> ReadEnums(object obj, BindTask task)
+                        {
+                            return new RenamedEntry(type.Original, "ErrorCodes");
+                        }
+                        if (typeStr.EndsWith(".flags"))
+                        {
+                            newName = typeStr.Substring(0, typeStr.Length - ".flags".Length);
+                            flagEnums.Add(newName);
+                            return new RenamedEntry(type.Original, newName);
+                        }
+                        newName = typeStr.Replace('.', '_');
+                        if (flagEnums.Contains(typeStr))
+                        {
+                            flagEnums.Add(newName);
+                        }
+                        return new RenamedEntry(type.Original, newName);
+                    }
+                );
+
+            apis.Elements("require")
+                .Elements("enum").ForEach(x =>
+                {
+                    var name = Rename(x.Attribute("name").Value, task);
+
+                    if (constants.Contains(name)) return;
+
+                    var type = Rename(x.Parent.Attribute("comment")?.Value, task);
+
+                    var typeStr = type?.Renamed;
+
+                    if (typeStr == null || typeStr.Contains(' '))
+                    {
+                        if (enumsPass1.TryGetValue(name, out var pass1type))
+                        {
+                            type = pass1type;
+                        }
+                        else
+                        {
+                            typeStr = name.Renamed switch
+                            {
+                                "CL_CONTEXT_MEMORY_INITIALIZE_KHR" => "cl_context_properties",
+                                "CL_DEVICE_PARTITION_BY_NAMES_INTEL" => "cl_device_partition_property",
+                                _ => typeStr
+                            };
+
+                            if (typeStr?.Contains("cl_device_info ") == true) typeStr = "cl_device_info";
+                            if (typeStr == "Error codes" || typeStr.StartsWith("Error type")) typeStr = "ErrorCodes";
+                            if (typeStr?.StartsWith("OpenCL ") == true && typeStr.Contains("deprecated"))
+                            {
+                                var typeNameStart = typeStr.IndexOf(' ', "OpenCL ".Length) + 1;
+                                var typeNameEnd = typeStr.IndexOf(' ', typeNameStart + 1);
+                                typeStr = typeStr.Substring(typeNameStart, typeNameEnd - typeNameStart);
+                            }
+                            if (typeStr?.StartsWith("Command type ") == true) typeStr = "cl_command_type";
+                            if (typeStr?.StartsWith("cl_uint ") == true) typeStr = typeStr.Substring("cl_uint ".Length);
+                            if (typeStr?.StartsWith("Additional ") == true) typeStr = typeStr.Substring("Additional ".Length);
```

Comment:
All in all I'm extremely uncomfortable using the "comment" attribute meaningfully. If there is some data in the comment attributes that we need, I'd recommend this be augmented & extracted to its own attribute in the upstream specs. Or at least this has been the approach with OpenGL.

Reply:
I do agree that it's not perfect but it's actually the last resort thing for stuff that's otherwise dangling.
For what it's worth, all this works back to Nov 2020 spec file, and the situation seems to be (slowly) improving.

Reply:
I still think we should add extra attributes to the upstream spec rather than try and dissect a field with no defined format. This is what we've done in the past with OpenGL (we added the `group` attribute).